functions/src/stories.ts

```diff
@@ -11,83 +11,161 @@ const index = algoliasearch(
+    path += "/translations/" + translationID;
+  }
+  return db.doc(path);
+}
+
 export const indexStories = functions
     .region("europe-central2")
     .firestore.document(
         "stories/{storyID}/translations/{translationID}/assets/story"
     )
     .onWrite(async (change, context) => {
+      if (!change.after.exists) return;
       const translationID = context.params.translationID;
       const storyID = context.params.storyID;
+      const {title, description, tags} = change.after.data()!;
+      const story = await storyDoc(storyID)
+          .get()
+          .then((doc) => doc.data()!);
+      const translation = await storyDoc(storyID, translationID)
+          .get()
+          .then((doc) => doc.data()!);
+
+      const entry: StoryRecord = {
+        storyID,
+        storyAuthorID: story.metaData.authorId,
+        translationID,
+        translationAuthorID: translation.metaData.authorId,
+        language: translation.language,
+        title,
+        description,
+        tags,
+      };
       if (change.before.exists) {
-        await index.deleteBy({
-          filters: "translationID:" + translationID,
-        });
-      }
-      if (change.after.exists) {
-        const {title, description, tags} = change.after.data()!;
-        const story = await db
-            .doc(`stories/${storyID}`)
-            .get()
-            .then((doc) => doc.data()!);
-        const translation = await db
-            .doc(`stories/${storyID}/translations/${translationID}`)
-            .get()
-            .then((doc) => doc.data()!);
+        await index.partialUpdateObject(
+            {objectID: translationID, ...entry},
+            {createIfNotExists: true}
+        );
+      } else {
         await index.saveObject(
-        {
-          storyID,
-          storyAuthorID: story.metaData.authorId,
-          translationID,
-          translationAuthorID: translation.metaData.authorId,
-          language: translation.language,
-          title,
-          description,
-          tags,
-          likes: translation.metaData.likes,
-        } as storyRecord,
-        {autoGenerateObjectIDIfNotExist: true}
+            entry,
+            {autoGenerateObjectIDIfNotExist: true}
         );
       }
     });
 
-// TODO
+export const trackStoryUpdateTime = functions
+    .region("europe-central2")
+    .firestore.document(
+        "stories/{storyID}/translations/{translationID}/assets/{assetID}"
+    )
+    .onWrite(async (change, context) => {
+      if (!change.after.exists) return;
+      await storyDoc(context.params.storyID, context.params.translationID)
+          .update({
+            "metaData.lastUpdateAt": firestore.FieldValue.serverTimestamp(),
+          });
+    });
+
+export const updateStoryMeta = functions
+    .region("europe-central2")
+    .firestore.document(
+        "stories/{storyID}/translations/{translationID}"
+    )
+    .onWrite(async (change, context) => {
+      if (change.after.exists) {
```

Comment:
You were using guard clauses everywhere, what happened to this one in particular? 😅 
```ts
if (!change.after.exists) return;
```

Reply:
Just thought the usual nested way is fine for those few lines. But for consistency, yes, might convert this as well.